code/datums/elements/tool_renaming.dm

```diff
@@ -0,0 +1,68 @@
+/**
+ * Renaming tool element
+ *
+ * When using this tool on an object with UNIQUE_RENAME,
+ * lets the user rename/redesc it.
+ */
+/datum/element/tool_renaming
+
+/datum/element/tool_renaming/Attach(datum/target)
+	. = ..()
+	if(!isitem(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_ITEM_INTERACTING_WITH_ATOM, PROC_REF(attempt_rename))
+
+/datum/element/tool_renaming/Detach(datum/source)
+	. = ..()
+	UnregisterSignal(source, COMSIG_ITEM_INTERACTING_WITH_ATOM)
+
+/datum/element/tool_renaming/proc/attempt_rename(datum/source, mob/living/user, atom/interacting_with, list/modifiers)
+	if(!isobj(interacting_with))
+		return NONE
+	
+	var/obj/renamed_obj = interacting_with
+
+	if(!(renamed_obj.obj_flags & UNIQUE_RENAME))
+		return NONE
+
+	INVOKE_ASYNC(src, PROC_REF(async_rename), user, renamed_obj)
+	return ITEM_INTERACT_SUCCESS
+
+/datum/element/tool_renaming/proc/async_rename(mob/living/user, obj/renamed_obj)
+	var/pen_choice = tgui_input_list(user, "What would you like to edit?", "Pen Setting", list(OPTION_RENAME, OPTION_DESCRIPTION, OPTION_RESET))
```

Comment:
```suggestion
	var/pen_choice = tgui_input_list(user, "What would you like to edit?", "Customization", list(OPTION_RENAME, OPTION_DESCRIPTION, OPTION_RESET))
```

pretty weird to have it be "Pen Setting" if this can be re-used on non-pen things. every other use of `pen` related stuff is fine in the backend imo but let's at least make sure the user-facing stuff is fine.

Reply:
Good catch
Also adjusted other pen stuff, no need for `pen_choice` to stay like that, and the pen label trait was only used by this systems predecessor before so, might as well rename it